rootfs/api/models/appsettings.py

```diff
@@ -86,6 +90,46 @@ def update_whitelist(self, previous_settings):
                     self.summary += ' and '
                 self.summary += "{} {}".format(self.owner, changes)
 
+    def update_autoscale(self, previous_settings):
+        data = getattr(previous_settings, 'autoscale', {}).copy()
+        new = getattr(self, 'autoscale', {}).copy()
+        # If no previous settings then do nothing
+        if not previous_settings:
+            return
+
+        # if nothing changed copy the settings from previous
+        if not new and data:
+            setattr(self, 'autoscale', data)
+        elif data != new:
+            for proc, scale in new.items():
+                if scale is None:
+                    # error if unsetting non-existing key
+                    if proc not in data:
+                        raise UnprocessableEntity('{} does not exist under {}'.format(proc, 'autoscale'))  # noqa
+                    del data[proc]
+                else:
+                    data[proc] = scale
```

Comment:
yes...but i think handling them is better asking user to enter the whole thing again...right now there are only 3 parameters but in future it might grow into more right?
can we override create method https://docs.djangoproject.com/en/1.10/ref/models/instances/#django.db.models.Model to copy previous into current if its get called before the serializers but am not sure if it works.